### PR TITLE
Portfolio  preserve data in the browser

### PR DESCRIPTION
**Pull request by TamaraDimov**
Implement the following interactions:

- When the user changes the content of any input field, the data is saved to the local storage.

- When the user loads the page, if there is any data in the local storage the input fields are pre-filled with this data.

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -220,21 +223,33 @@ window.addEventListener('load', () => {
...
     document.getElementById('contact-form').submit();
   });
+
+  const contactFormData = JSON.parse(
+    localStorage.getItem('contactform') ?? '{}',
+  );
+  name.value = contactFormData.name ?? '';
+  email.value = contactFormData.email ?? '';
+  text.value = contactFormData.message ?? '';
 });
```
Comment (Munsa1) [inline script.js:255]: - [ ] Hi, kindly add an on load function which pulls the data from local storage and populates it into the input fields upon reloading the page. In the current scenario the data is not displayed

File: script.js [hunk 2]
```diff
@@ -220,21 +223,33 @@ window.addEventListener('load', () => {
...
       document.getElementById('error').innerHTML = 'Invalid Text';
       e.preventDefault();
       return;
     }
-    // show errors
+
+    const form = document.getElementById('contact-form');
+    const myFormData = new FormData(form);
+    const formDataObj = Object.fromEntries(myFormData.entries());
+    localStorage.setItem('contactform', JSON.stringify(formDataObj));
```
Comment (Munsa1) [inline script.js:244]: - [ ] When the user enter values into the input fields, the form data for all inputs should be saved to a single local storage registry, which currently is not the case. Kindly fix this

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @TamaraDimov, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: script.js (hunk 1 above)
Comment (mrprotocoll) [reply script.js:255]: Thank you @Munsa1. 

Do you mean we should put the code above in an onload event? if yes, that is what we have done, and the data actually populates when you reload the page after filling the form.

could you clarify more on what we need to do?

File: script.js (hunk 2 above)
Comment (mrprotocoll) [reply script.js:244]: Thank You @Munsa1. We will correct that

**Comment by alexander16108** [comment]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**